### PR TITLE
New section "architectural storage" in chapter 3.

### PR DESCRIPTION
**Pull request by NicholasWoodIMG**

### CODE REVIEW

File: specification/src/chapter3.adoc [hunk 1]
```diff
@@ -450,6 +450,80 @@ and vector cryptography extensions.
...
+| Cryptographically protected architectural metadata storage MUST provide integrity protection
+
+| SR_AMS_007
+| Cryptographically protected architectural metadata storage SHOULD provide replay protection
+
+|===
+
+Architectural metadata needs to be protected against both unauthorized access, and errors (accidental or malicious).
+
+On systems where architectural metadata is stored in external memory, and external memory attacks are in scope (for example, directly accessible or replacable external memory), then cryptographic replay protection is strongly recommended.
```
Comment (lfiolhais) [inline specification/src/chapter3.adoc:512]: Typo in `replacable`

**Comment by lfiolhais** [comment]
Why is the title of the section "metadata storage"? The guidelines provided in the section refer to best practices for any sensitive data. In a general sense, these recommendations need to be applied to any piece of data that is implicitly trusted by the architecture in a secure application/environment.

File: specification/src/chapter3.adoc (hunk 1 above)
Comment (andrewdellow) [reply specification/src/chapter3.adoc:512]: 'access' here includes reading and modifying ? 

also need to consider data relocation attacks when stored in external memory.  encryption must be location dependent, freshness protected, and authenticated.

Comment (NicholasWoodIMG) [reply specification/src/chapter3.adoc:512]: Fixed "replaceable".

Added location freshness to SR_AMS_005.

Added temporal freshness to SR_AMS_006.

Clarified the rationale paragraphs.

**Comment by NicholasWoodIMG** [comment]
> Why is the title of the section "metadata storage"? The guidelines provided in the section refer to best practices for any sensitive data. In a general sense, these recommendations need to be applied to any piece of data that is implicitly trusted by the architecture in a secure application/environment.

The section title is "architectural metadata storage".

I updated the definitions in the intro title to reflect your comments.

**Review by andrewdellow — APPROVED** [review]
looks good

**Review by elliops — APPROVED** [review]
Looks good

**Pull request merged**